src/App.tsx

```diff
@@ -15,58 +16,67 @@ export const goods = [
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [good, SetGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      {!good ? (
+        <h1 className="title">No goods selected</h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {`${good} is selected`}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              SetGood('');
+            }}
+            aria-label="clear"
+          />
+        </h1>
+      )}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+      <table className="table">
+        <tbody>
+          {goods.map(initialGood => {
+            return (
+              <tr
+                key={initialGood}
+                data-cy="Good"
+                className={classNames({
+                  'has-background-success-light': good === initialGood,
+                })}
+              >
+                <td>
+                  <button
+                    data-cy={
+                      good === initialGood ? 'RemoveButton' : 'AddButton'
+                    }
+                    type="button"
+                    className={
+                      good === initialGood ? 'button is-info' : 'button'
+                    }
+                    onClick={() => {
+                      if (good === initialGood) {
+                        SetGood('');
+                      } else {
+                        SetGood(initialGood);
+                      }
+                    }}
```

Comment:
Create separate methods instead of creating inline event handlers (if we do so - we don't create new arrow function after component re-rendering) [[checklist]](https://github.com/mate-academy/react_goods-selector/blob/master/checklist.md)